sdk/eventhub/ci.legacy.yml

```diff
@@ -0,0 +1,55 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
+
+trigger:
+  branches:
+    include:
+    - master
+    - hotfix/*
+    - release/*
+  paths:
+    include:
+    - sdk/eventhub/
+    exclude:
+    - sdk/eventhub/Azure.Messaging.EventHubs
+    - sdk/eventhub/Azure.Messaging.EventHubs.Processor
+    - sdk/eventhub/Azure.Messaging.EventHubs.Shared
+    - sdk/eventhub/Azure.ResourceManager.EventHubs
+    - sdk/eventhub/Microsoft.Azure.Management.EventHub
+    - sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs
+
+pr:
+  branches:
+    include:
+    - master
+    - feature/*
+    - hotfix/*
+    - release/*
+  paths:
+    include:
+    - sdk/eventhub/
+    exclude:
+    - sdk/eventhub/Azure.Messaging.EventHubs
```

Comment:
I'm a bit concerned with having to keep track of the projects/paths in multiple places and keep it all in sync without making mistakes. Is there maybe some import voodoo we can use to centralize this information or write a script that would make sure things are up-to-date and correct?

Reply:
These particular triggers are owned by DevOps, unfortunately, and aren't something that we have much influence over in terms of placement and syntax.    We could use a sub template to import them into a variable, but this set is used in a single place, so there's little benefit.  

The `ExcludeProject` parameter that we added has a different format with the goal of less ceremony and more flexibility.  It's meant to be more of a "tell me the pattern of project names that you don't care about and we'll ignore anything under that."   With the wildcard support, it shouldn't be necessary to change that often nor adapt to most changes.

To your point, the best way to centralize would likely be a script that could accept the wildcard patterns and generate the corresponding `exclude` paths.  I think that's probably a bit of a premature optimization at this point, as there aren't many places where the T1 and T2 code is bound together  under different ownership.  For those that are, the T1 assets are in maintenance mode and it isn't likely that we'll see path changes.  If we start to see people splitting pipelines in areas that are more volatile, I think that is a great time to revisit.

Reply:
I'm going to commit this in the current form for now, to unblock me from splitting the Messaging libraries out.   I'll file an issue to track considering a script to automate at a later time.

Reply:
Yes explicitly listing and maintaining these include/exclude paths has annoyed me as well. Unfortunately it is a devops limitation currently and so we just have to pay the cost until we can find a better solution.

I will say I do generally try and avoid mixing include and exclude path filters. I generally prefer we explicitly have an include list for the paths we want to trigger this. That way it is clearer what we expect to trigger this pipeline and only the people that care about this pipeline will need to edit it, as opposed to other folks that don't care about it but have to keep adding excludes to it. Of course another issue that will likely occur is that when a new directory is added someone will have to remember to add the path otherwise no pipeline will trigger.

Reply:
For this, I was following the [example](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/ci.data.yml) that had been shared with me during the planning for this.  I'd prefer to have the inclusion by default with the rationale that its better to inadvertently include something that you need to exclude later than to forget to add something and not realize that it isn't included in CI.